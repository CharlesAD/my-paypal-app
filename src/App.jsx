import "./App.css";
import { useState } from "react";

function App() {
  const [message, setMessage] = useState("What would you like to send?");
  const [number, setNumber] = useState(0);

  const balance = 100;

  const newBalance = () => {
    if (number > balance) {
      setMessage("You do not have enough funds");
      console.log("broke");
    } else {
      setMessage("Thank you for your transfer");
      console.log("success");
    }
  };

  const newValue = (event) => {
    setNumber(event.target.valueAsNumber);
  };

  return (
    <>
      <div className="text-center">
        <header className="">
          <h1>Welcome to My Piggy Bank</h1>
        </header>
      </div>
      <div>
        <h2>Your balance is {balance} </h2>
        <label htmlFor="transfer">How much would you like to transfer?</label>
        <input
          type="number"
          placeholder="Enter your desired transfer balance"
          id="transfer"
          name="transfer"
          onChange={newValue}
        />
      </div>
      <button
        className="bg-green-600"
        type="submit"
        label="send"
        id="send"
        onClick={newBalance}
      >
        Send
      </button>
      <label htmlFor="feedback" hidden>
        Feedback
      </label>
      <p id="feedback" name="feedback">
        {message}
      </p>
    </>
  );
}

export default App;
