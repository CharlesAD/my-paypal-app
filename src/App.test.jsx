import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

it("renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("calls the newBalance function whenever the button is clicked", () => {
  const mockSend = jest.fn();

  render(<App newBalance={mockSend} />);

  userEvent.click(screen.getByRole("button"));

  expect(mockSend).toHaveBeenCalled();
});

it("displays 'You do not have enough funds' when transfer value is greater than balance ", () => {
  render(<App />);

  const enteredValue = screen.getByLabelText(/transfer/i);

  const sendBtn = screen.getByRole("button", { name: /send/i });

  userEvent.click(sendBtn);

  const displayMessage = screen.getByLabelText(/feedback/i);

  expect(displayMessage).toBe("You do not have enough funds");
});
