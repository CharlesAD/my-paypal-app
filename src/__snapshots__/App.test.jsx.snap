// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div
    className="text-center"
  >
    <header
      className=""
    >
      <h1>
        Welcome to My Piggy Bank
      </h1>
    </header>
  </div>,
  <div>
    <h2>
      Your balance is 
      100
       
    </h2>
    <label
      htmlFor="transfer"
    >
      How much would you like to transfer?
    </label>
    <input
      id="transfer"
      name="transfer"
      onChange={[Function]}
      placeholder="Enter your desired transfer balance"
      type="number"
    />
  </div>,
  <button
    className="bg-green-600"
    id="send"
    label="send"
    onClick={[Function]}
    type="submit"
  >
    Send
  </button>,
  <label
    hidden={true}
    htmlFor="feedback"
  >
    Feedback
  </label>,
  <p
    id="feedback"
    name="feedback"
  >
    What would you like to send?
  </p>,
]
`;
